graphql_compiler/tests/test_schema_generation.py

```diff
@@ -181,10 +221,46 @@ def test_link_property(self):
         self.assertEqual(out_property.qualifier, 'Person')
 
     def test_parsed_class_fields(self):
-        schema_data = [BASE_VERTEX_SCHEMA_DATA, ENTITY_SCHEMA_DATA,
-                       BASE_EDGE_SCHEMA_DATA, PERSON_SCHEMA_DATA,
-                       LOCATION_SCHEMA_DATA, PERSON_LIVES_IN_EDGE_SCHEMA_DATA]
+        schema_data = [
+            BASE_EDGE_SCHEMA_DATA,
+            BASE_VERTEX_SCHEMA_DATA,
+            ENTITY_SCHEMA_DATA,
+            LOCATION_SCHEMA_DATA,
+            PERSON_LIVES_IN_EDGE_SCHEMA_DATA,
+            PERSON_SCHEMA_DATA,
+        ]
         schema_graph = SchemaGraph(schema_data)
         person_lives_in_edge = schema_graph.get_element_by_class_name('Person_LivesIn')
         self.assertEqual(PERSON_LIVES_IN_EDGE_SCHEMA_DATA['customFields'],
                          person_lives_in_edge.class_fields)
+
+    def test_type_equivalence_dicts(self):
+        schema_data = [
+            BASE_EDGE_SCHEMA_DATA,
+            BASE_VERTEX_SCHEMA_DATA,
+            BABY_SCHEMA_DATA,
+            ENTITY_SCHEMA_DATA,
+            LOCATION_SCHEMA_DATA,
+            PERSON_LIVES_IN_EDGE_SCHEMA_DATA,
+            PERSON_SCHEMA_DATA,
+        ]
+        schema, type_equivalence_dicts = get_graphql_schema_from_orientdb_schema_data(schema_data)
+        person, person_baby_union = next(six.iteritems(type_equivalence_dicts))
```

Comment:
Doesn't this implicitly depend on ordering of a dict, which is undefined except for one-element dicts? Since we know the key we're looking for, why not just look it up and assert that it exists?

Also, since we're trying to test `type_equivalence_hints`, shouldn't we ensure that the detected hints are exactly as we expect? Right now, we're checking a single key, but we don't e.g. make sure that's the only key in there (assuming that's what we expected).